### PR TITLE
Tweak task for Indexed completion Job example

### PR DESCRIPTION
**Pull request by sftim**
A few changes to round off the new task added in PR #26958

/kind cleanup
/sig apps
/milestone 1.21
/priority backlog

/cc alculquicondor

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Thanks

/lgtm

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md [hunk 1]
```diff
@@ -26,14 +26,14 @@ expose the index in the `JOB_COMPLETION_INDEX` environment variable.
...
 Here is an overview of the steps in this example:
 
-1. **Create an image that can read the pod index**. You might modify the worker
-   program or add a script wrapper.
-2. **Start an Indexed Job**. The downward API allows you to pass the annotation
-   as an environment variable or file to the container.
+1. **Define a Job manifest using indexed completion**.
+   The downward API allows you to pass the pod index annotation as an
+   environment variable or file to the container.
+2. **Start an `Indexed` Job based on that manifest**.
```
Comment (kbhawkey) [inline content/en/docs/tasks/job/indexed-parallel-processing-static.md:32]: 👋 @sftim ,

Is there a reason to use bold and code formatting in the instructions?
An indexed Job is a `Job`?

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md [hunk 2]
```diff
@@ -66,13 +66,21 @@ program accepts a file as an argument and prints its content reversed.
 rev data.txt
 ```
 
-For this example, you'll use the `rev` tool from the
+You'll use the `rev` tool from the
```
Comment (kbhawkey) [inline content/en/docs/tasks/job/indexed-parallel-processing-static.md:69]: Line 69 seems confusing. Could you combine that sentence with lines 61-63?

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md [hunk 3]
```diff
@@ -66,13 +66,21 @@ program accepts a file as an argument and prints its content reversed.
...
 [`busybox`](https://hub.docker.com/_/busybox) container image.
 
+As this is only an example, each Pod only does a tiny piece of work (reversing a short
+string). In a real workload you might, for example, create a Job that represents
+ the
```
Comment (kbhawkey) [inline content/en/docs/tasks/job/indexed-parallel-processing-static.md:74]: nit: remove the empty spaces in line 74

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md [hunk 4]
```diff
@@ -66,13 +66,21 @@ program accepts a file as an argument and prints its content reversed.
...
+task of producing 60 seconds of video based on scene data.
+Each work item in the video rendering Job would be to render a particular
```
Comment (kbhawkey) [inline content/en/docs/tasks/job/indexed-parallel-processing-static.md:76]: nit: reword
Each work item in the video rendering Job renders a particular frame ...

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md [hunk 5]
```diff
@@ -66,13 +66,21 @@ program accepts a file as an argument and prints its content reversed.
...
+frame of that video clip. Indexed completion would mean that each Pod in
+the Job knows which frame to render and publish, by counting frames from
+the start of the clip.
+
 ## Define an Indexed Job
 
-Here is a job definition. You'll need to edit the container image to match your
-preferred registry.
+Here is a sample Job manifest that uses `Indexed` completion mode:
```
Comment (kbhawkey) [inline content/en/docs/tasks/job/indexed-parallel-processing-static.md:83]: nit: What is "Indexed" completion mode? Should this be capitalized and use code formatting?

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md [hunk 6]
```diff
@@ -96,10 +104,15 @@ like shown in the following example:
 Now run the Job:
 
 ```shell
-kubectl apply -f ./indexed-job.yaml
+# This uses the first approach (relying on $JOB_COMPLETION_INDEX)
+kubectl apply -f https://kubernetes.io/examples/application/job/indexed-job.yaml
 ```
 
-Wait a bit, then check on the job:
+When you create this Job, the control plane creates a series of Pods, one for each index you specified. The value of `.spec.parallelism` determines how many can run at once whereas `.spec.completions` determines how many Pods the Job creates in total.
```
Comment (kbhawkey) [inline content/en/docs/tasks/job/indexed-parallel-processing-static.md:111]: nit: The value of ... determines how many Pods (?) can run [immediately or at the same time]?

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/tasks/job/indexed-parallel-processing-static.md:32]: The formatting here is as per the original.

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/tasks/job/indexed-parallel-processing-static.md:69]: We could omit this paragraph - @alculquicondor is that OK?

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md (hunk 5 above)
Comment (sftim) [reply content/en/docs/tasks/job/indexed-parallel-processing-static.md:83]: The page prerequisites direct the reader to read up about Job first, which (in the v1.21 docs) explains that mode. We could link directly to that part of the page to emphasize this?

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md (hunk 1 above)
Comment (alculquicondor) [reply content/en/docs/tasks/job/indexed-parallel-processing-static.md:32]: An Indexed Job is a Job with `.spec.completionMode=Indexed`. Bold could be omitted. I just think it makes it easier to get the gist of the steps.

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md (hunk 2 above)
Comment (alculquicondor) [reply content/en/docs/tasks/job/indexed-parallel-processing-static.md:69]: SGTM

File: content/en/docs/tasks/job/indexed-parallel-processing-static.md (hunk 5 above)
Comment (alculquicondor) [reply content/en/docs/tasks/job/indexed-parallel-processing-static.md:83]: I don't think it's necessary. The following paragraph is the example, which shows how the completion mode is set in the spec.

**Comment by kbhawkey** [comment]
Since this is for 1.21,
/approve

**Pull request merged**